### PR TITLE
adding a flag for the status of the dl1 extraction using the lhfit

### LINKED ISSUE #6: Failed event classifier
When using the lhfit feature for determination of DL1 parameters, some events are not processed by the associated function.
After the first extraction of DL1 parameter using the Hillas method, the get_dl1_lhfit function is not called for one of 3 reasons :
- the number of pixels surviving the cleaning is zero -- expected behavior, the dl1 parameters are also not extracted with the Hillas method
-  the number of pixels surviving the cleaning is above 1000 -- this is a temporary fix to avoid flasher events on data but it can also remove some useful events
- The image contains pixels with high luminosity, which would require to compute the likelihood function considering a large range of potential number of photon electrons (each being a term in a theoretically infinite sum). The issue being an associated large computational time.

We propose the implementation of a flag in the data identifying if an event was or not processed by the lhfit function, and if not the reason why.

Later, improvement of the handling of high luminosity events should remove the need for conditions stopping the processing (issue #7 ). Flasher events may still need to be identified.

### PR DESCRIPTION
**Pull request by gabemery**
## Description
Adds a new flag to the lstchain dl1 container. The flag gives for each event the status of the processing by the LH fit method.
Default value is "None".

## Related Issue
Closes #6

## How Has This Been Tested?
Not tested

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by mexanick** [comment]
> Hello @gabemery! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
> * In the file [`lstchain/io/lstcontainers.py`](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/io/lstcontainers.py):
> 
> > [Line 119:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/io/lstcontainers.py#L119): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> 
> * In the file [`lstchain/reco/r0_to_dl1.py`](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py):
> 
> > [Line 629:28](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L629): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> > [Line 630:29](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L630): [E117](https://duckduckgo.com/?q=pep8%20E117) over-indented
> > [Line 630:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L630): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> > [Line 632:49](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L632): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> > [Line 632:50](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L632): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> > [Line 632:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L632): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (191 > 79 characters)
> > [Line 633:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L633): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> > [Line 639:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L639): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> > [Line 640:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L640): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> > [Line 641:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L641): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> > [Line 642:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L642): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> > [Line 645:80](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L645): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> > [Line 646:25](https://github.com/cta-sst-1m/cta-lstchain/blob/52170c1690989a9bcd6a75b6725219a13511c2c3/lstchain/reco/r0_to_dl1.py#L646): [E901](https://duckduckgo.com/?q=pep8%20E901) IndentationError: unindent does not match any outer indentation level

@gabemery please make the PEP8 bot happy ;) Especually the last reported issue

File: lstchain/io/lstcontainers.py [hunk 1]
```diff
@@ -116,6 +116,8 @@ class DL1ParametersContainer(Container):
     ucts_trigger_type = Field(None, "UCTS trigger type")
     trigger_time = Field(None, "trigger time")
 
+    lhfit_call_status(None, "Status of the processing of the event by the LH fit method")
```
Comment (mexanick) [inline lstchain/io/lstcontainers.py]: How this is supposed to work? If `lhfit_call_status` is a field, then it should be declared as other fields above. i.e.
```python
lhfit_call_status = Field(None, "Status of the processing of the event by the LH fit method")
```

File: lstchain/reco/r0_to_dl1.py [hunk 2]
```diff
@@ -624,22 +624,29 @@ def r0_to_dl1(
...
-                                                        normalized_pulse_template=pulse_template,
-                                                        dl1_container=dl1_filled,
-                                                        custom_config=config,
-                                                        use_main_island=True)
+                    if ('lh_fit_config' in config.keys():
+                        if dl1_filled['n_pixels'] is not 0
+                           and dl1_filled['n_pixels'] < 1000):
+                            is_saturated = np.any(image > config['lh_fit_config']['n_peaks'])
+
+                            if not is_saturated: #reject computationnally expensive events which would be poorly estimate with the selected value of n_peak TODO : improve to not reject events
```
Comment (mexanick) [inline lstchain/reco/r0_to_dl1.py]: Please abide the PEP8 rules for comment strings

File: lstchain/reco/r0_to_dl1.py [hunk 3]
```diff
@@ -624,22 +624,29 @@ def r0_to_dl1(
...
+                                dl1_filled.lhfit_call_status = "Not processed : Error in function"
```
Comment (mexanick) [inline lstchain/reco/r0_to_dl1.py]: If you envision a possibility of failure, the next code block with a call to `get_dl1_lh_fit` function has to be wrapped into try-except block

**Review by mexanick — CHANGES_REQUESTED** [review]

File: lstchain/io/lstcontainers.py (hunk 1 above)
Comment (gabemery) [reply lstchain/io/lstcontainers.py]: Thanks for noticing. It is indeed an overview

File: lstchain/reco/r0_to_dl1.py (hunk 3 above)
Comment (gabemery) [reply lstchain/reco/r0_to_dl1.py]: Not sure about the except argument that I used, but it seems quite generic. From ctapipe:
`
class HillasParameterizationError(RuntimeError):
    pass
`

File: lstchain/reco/r0_to_dl1.py [hunk 4]
```diff
@@ -618,28 +619,50 @@ def r0_to_dl1(
...
+                                                                subarray,
+                                                                telescope_id,
+                                                                image=image,
+                                                                is_simu=is_simu,
+                                                                normalized_pulse_template=pulse_template,
+                                                                dl1_container=dl1_filled,
+                                                                custom_config=config,
+                                                                use_main_island=True)
+                                    dl1_filled.lhfit_call_status = "Processed"
+                                except HillasParameterizationError:
```
Comment (mexanick) [inline lstchain/reco/r0_to_dl1.py]: Please check the code and see which kind of errors in can throw. There'll be probably `HillasParametrizationError` from the regular `get_dl1` call plus possible errors from LH-fit code itself

File: lstchain/reco/r0_to_dl1.py (hunk 3 above)
Comment (mexanick) [reply lstchain/reco/r0_to_dl1.py]: see my other comment

**Review by mexanick — APPROVED** [review]

**Pull request merged**